bundles/org.openhab.binding.omnilink/src/main/java/org/openhab/binding/omnilink/internal/discovery/OmnilinkDiscoveryService.java

```diff
@@ -405,27 +406,22 @@ private void discoverThermostats() {
 
                 Map<String, Object> properties = Map.of(THING_PROPERTIES_NAME, thingName);
 
-                final SystemType systemType = this.systemType;
-                if (systemType != null) {
-                    switch (systemType) {
-                        case LUMINA:
-                            thingUID = new ThingUID(THING_TYPE_LUMINA_AREA, bridgeUID, thingID);
-                            break;
-                        case OMNI:
-                            thingUID = new ThingUID(THING_TYPE_OMNI_AREA, bridgeUID, thingID);
-                            break;
-                        default:
-                            throw new IllegalStateException("Unknown System Type");
-                    }
+                switch (systemType.get()) {
```

Comment:
This will throw an unchecked `NoSuchElementException` if the Optional is empty. You could encapsulate it with `systemType.ifPresent(t -> ...)`.

Reply:
@fwolter I apologize but I am really new to lambda functions in Java, I see the issue here but I'm not sure how to use the example you gave to solve it.

Reply:
The code would look like this:

`systemType.ifPresent(t -> {
    switch(t) {
        case ...:
        ....
    }
});`

Reply:
OK, there is one problem with that, Java throws and error because of a `Local variable 'varName' defined in an enclosing scope must be final or effectively final` see [here](https://www.yawintutor.com/local-variable-name-defined-in-an-enclosing-scope-must-be-final-or-effectively-final/) for more details on the exception.

Reply:
`thingUID` needs to be declared within the lambda expression.